Tensile/Components/MAC_F32.py

```diff
@@ -67,31 +72,86 @@ def __call__(self, writer, m, innerUnroll):
-                        kStr += "s_barrier // extra barrier for performance{endLine}".format_map(vars)
-                    macIdx += 1
+        if dualMacEnable == 1:
+            instQ = queue.Queue();
+            for iui in range(0, innerUnroll):
+               for idx1 in range(0, kernel["ThreadTile1"]):
+                   for idx0 in range(0, kernel["ThreadTile0"]):
+                        vars["idx0"] = idx0
+                        vars["idx1"] = idx1
+                        vars["a"] = idx0 if writer.tPB["tile01Idx"] else idx1
+                        vars["b"] = idx1 if writer.tPB["tile01Idx"] else idx0
+                        vars["iui"] = iui
+
+                        vars["cStr"] = "v[vgprValuC + {idx0} + {idx1}*{ThreadTile0}]".format_map(vars)
+                        vars["aStr"] = "v[vgprValuA_X{m}_I{iui} + {a}]".format_map(vars)
+                        vars["bStr"] = "v[vgprValuB_X{m}_I{iui} + {b}]".format_map(vars)
+
+                        instVars = {}
+                        instVars["endLine"] = writer.endLine
+                        instVars["cStr"] = vars["cStr"]
+                        instVars["aStr"] = vars["aStr"]
+                        instVars["bStr"] = vars["bStr"]
+                        if instQ.empty():
+                            instQ.put(instVars)
+                        else:
+                            # pop instruction
+                            prevVars = instQ.queue[0]
+                            if instVars["cStr"] != prevVars["cStr"]:
+                                # make dual fmac
+                                kStr += "v_dual_fmac_f32 {cStr}, {aStr}, {bStr}".format_map(prevVars) + " :: v_dual_fmac_f32 {cStr}, {aStr}, {bStr}{endLine}".format_map(vars)
+                                kStr += priority(writer, 1, "Raise priority while processing macs")
+                                instQ.get()
+                            else:
+                                # push instruction
+                                instQ.put(instVars)
+
+                        if macIdx == kernel["PerformanceWaitLocation"]:
+                            # pop all instructions
+                            while instQ.qsize() > 0:
+                                prevVars = instQ.get()
+                                kStr += "{instruction} {cStr}, {aStr}, {bStr}{endLine}".format_map(prevVars)
+                                kStr += priority(writer, 1, "Raise priority while processing macs")
+                            kStr += "s_waitcnt lgkmcnt({PerformanceWaitCount}) // extra wait for performance{endLine}".format_map(vars)
+                        if macIdx == kernel["PerformanceSyncLocation"]:
+                            # pop all instructions
+                            while instQ.qsize() > 0:
+                                prevVars = instQ.get()
+                                kStr += "v_fmac_f32 {cStr}, {aStr}, {bStr}{endLine}".format_map(prevVars)
```

Comment:
This should be instruction instead of using fixed "v_fmac_f32".
If we would like to always use "v_fmac_f32". we should set instruction = "v_fmac_f32" and check asmCap before using it.